src/Cake.Common/Tools/GitLink/GitLinkRunner.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+using Cake.Core.IO;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.GitLink
+{
+    /// <summary>
+    /// GitLink runner
+    /// </summary>
+    public sealed class GitLinkRunner : Tool<GitLinkSettings>
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly ICakeEnvironment _environment;
+        private readonly ICakeLog _log;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GitLinkRunner"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="globber">The globber.</param>
+        /// <param name="log">The logger.</param>
+        public GitLinkRunner(IFileSystem fileSystem, ICakeEnvironment environment, IProcessRunner processRunner, IGlobber globber, ICakeLog log)
+            : base(fileSystem, environment, processRunner, globber)
+        {
+            _fileSystem = fileSystem;
+            _environment = environment;
+            _log = log;
+        }
+
+        /// <summary>
+        /// Update pdb files to link all sources.
+        /// </summary>
+        /// <param name="solutionPath">The directory containing the solution with the pdb files.</param>
+        /// <param name="settings">The settings.</param>
+        public void Run(DirectoryPath solutionPath, GitLinkSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            if (solutionPath == null)
+            {
+                throw new ArgumentNullException("solutionPath");
+            }
+
+            Run(settings, GetArguments(solutionPath, settings));
+        }
+
+        private ProcessArgumentBuilder GetArguments(DirectoryPath solutionPath, GitLinkSettings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
+
+            builder.AppendQuoted(solutionPath.MakeAbsolute(_environment).FullPath);
+
+            if (settings.RepositoryUrl != null)
```

Comment:
Agree when you need to code against it, but this is just a pass-through object anyway. You even have chance it will "mess" (change) your stuff this way.